crates/runtime/tests/models/hf.rs

```diff
@@ -118,142 +117,104 @@ async fn huggingface_test_search() -> Result<(), anyhow::Error> {
+                });
 
-            let response = send_nsql_request(
-                http_base_url.as_str(),
-                "how many records in taxi_trips dataset?",
-                Some("hf_model"),
-                Some(true),
-                None,
-            )
-            .await?;
+                let llm_init_lock = LOCAL_LLM_INIT_MUTEX.lock().await;
 
-            assert!(
-                json_is_single_row_with_value(&response, 10),
-                "Expected a single record containing the value 10"
-            );
-
-            // ensure all spans are exported into task_history
-            let _ = trace_provider.force_flush();
-
-            let tasks = get_executed_tasks(&rt, task_start_time.into()).await?;
-
-            let sample_data_task = tasks
-                .iter()
-                .find(|t| t.0 == "tool_use::sample_data")
-                .expect("Expected 'tool_use::sample_data' task to be executed");
-
-            insta::assert_snapshot!("nsql_sample_data_task", sample_data_task.1);
+                tokio::select! {
+                    // increased timeout to download and load huggingface model
+                    () = tokio::time::sleep(std::time::Duration::from_secs(300)) => {
+                        return Err(anyhow::anyhow!("Timed out waiting for components to load"));
+                    }
+                    () = rt.load_components() => {}
+                }
 
-            let sql_query_task = tasks
-                .iter()
-                .find(|t| t.0 == "sql_query")
-                .expect("Expected 'sql_query' task to be executed");
+                drop(llm_init_lock);
+
+                runtime_ready_check(&rt).await;
+
+                let test_cases = [
+                    TestCase {
+                        name: "hf_with_model",
+                        body: json!({
+                            "query": "how many records (as 'total_records') are in taxi_trips dataset?",
+                            "model": "hf_model",
+                            "sample_data_enabled": false,
+                        }),
+                    },
+                    TestCase {
+                        name: "hf_with_sample_data_enabled",
+                        body: json!({
+                            "query": "how many records (as 'total_records') are in taxi_trips dataset?",
+                            "model": "hf_model",
+                            "sample_data_enabled": true,
+                        }),
+                    },
+                ];
+
+                for ts in test_cases {
+                    run_nsql_test(http_base_url.as_str(), &ts, &trace_provider).await?;
+                }
 
-            insta::assert_snapshot!("nsql_sample_data_task_sql_query", sql_query_task.1);
+                tracing::info!("/v1/nsql: Ensure error when invalid dataset name is provided");
+                assert!(send_nsql_request(
```

Comment:
Why can't these be test cases like above?

Reply:
It's because we expect an error? I wonder if we can account for that in `run_nsql_test` by not erroring on failed response [here](https://github.com/spiceai/spiceai/pull/3727/files?diff=split&w=0#diff-a56a01a97a0e053ada4281ce00dcfab87c5af4aa04bf3c0b58771e0730366168R72).

Reply:
Thank you - very good suggestion, updated